.github/workflows/nightly-build.yml

```diff
@@ -42,3 +42,41 @@ jobs:
 
       - name: Upload to S3
         run: aws s3 cp build/bin s3://amazoncloudwatch-agent/nightly-build/latest/ --recursive --acl public-read --source-region us-east-1
+
+      # Users should only take the tar of the darwin build according to our readme so deleting this does not matter. More of a sanity step
+      - name: Delete Mac Binary As This Should Only Be Built On Mac
+        run: |
+          aws s3 rm --recursive s3://amazoncloudwatch-agent/nightly-build/latest/darwin_amd64/
+          aws s3 rm --recursive s3://amazoncloudwatch-agent/nightly-build/latest/darwin_arm64/
+
+  mac-build:
+    name: Upload Mac Nightly Binaries
+    runs-on: macos-11
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ~1.19.2
+        id: go
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.S3_AWS_ASSUME_ROLE }}
+          aws-region: us-east-1
+
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Mac nightly release
+        run: make nightly-release-mac
+
+      - name: Upload to S3
```

Comment:
In the future, we can edit our build pkg script and upload that as well. I would rather not do it as part of this pr since it means changing 2 repo and more code changes. It is also not required atm.